File: src/backtracking/cpp/string_pattern/test/test_string_pattern.cpp
#include "string_pattern.h"
#include "gtest/gtest.h"

TEST(TestStringPattern, TestPositive1) {
  std::string string = "basetestcasebase";
  std::string pattern = "ABCDA";
  EXPECT_TRUE(string_pattern(string, pattern));
}

TEST(TestStringPattern, TestPositive2) {
  std::string string = "basetestcasebase";
  std::string pattern = "A";
  EXPECT_TRUE(string_pattern(string, pattern));
}

TEST(TestStringPattern, TestPositive3) {
  std::string string = "basetestcasebase";
  std::string pattern = "ABA";
  EXPECT_TRUE(string_pattern(string, pattern));
}

TEST(TestStringPattern, TestNegative1) {
  std::string string = "basetestcasebase";
  std::string pattern = "AA";
  EXPECT_FALSE(string_pattern(string, pattern));
}

TEST(TestStringPattern, TestNegative2) {
  std::string string = "abcdefg";
  std::string pattern = "aa";
  EXPECT_FALSE(string_pattern(string, pattern));
}

File: src/backtracking/python/string_pattern/tests/test_string_pattern.py
import unittest
from src.string_pattern import string_pattern


class TestStringPattern(unittest.TestCase):
    def test_positive_1(self):
        string = "basetestcasebase"
        pattern = "ABCDA"
        self.assertTrue(string_pattern(string, pattern))

    def test_positive_2(self):
        string = "basetestcasebase"
        pattern = "A"
        self.assertTrue(string_pattern(string, pattern))

    def test_positive_3(self):
        string = "basetestcasebase"
        pattern = "ABA"
        self.assertTrue(string_pattern(string, pattern))

    def test_negative_1(self):
        string = "basetestcasebase"
        pattern = "AA"
        self.assertFalse(string_pattern(string, pattern))

    def test_negative_2(self):
        string = "abcdefg"
        pattern = "aa"
        self.assertFalse(string_pattern(string, pattern))


if __name__ == "__main__":
    unittest.main()
